### PR TITLE
[PostEmscripten] Fix calcSegmentOffsets for large offsets

### PR DESCRIPTION
**Pull request by sbc100**
Specifically offsets larger than 2^32 which were being interpreted misinterpreted here as very large int64_t values.

### CODE REVIEW

**Comment by sbc100** [comment]
I'm not sure it worth writing a test for this.. maybe you can suggest a way if you feel like it needs one.

This bug is currently preventing some of the "2gb" tests in emscripten from running (where static data segments are >2Gb in the 32-bit space)

**Review by kripken — COMMENTED** [review]
In wasm64 don't these need to be 64-bit?

**Comment by sbc100** [comment]
> In wasm64 don't these need to be 64-bit?

The Address being assigned to here is 64-bit or 32-bit accordingly, that is not the problem..

The bug being fixed here only occurs when on wasm32 when the value being read is is > 2&32.  

Prior to this change the `getInteger()` would interpret this larger 32-bit value as a negative 32-bit value, which is wrong.  Data segment offsets should be interpreted as unsigned values.. which I think is exactly what this change does.

File: src/passes/PostEmscripten.cpp [hunk 1]
```diff
@@ -317,7 +316,7 @@ struct PostEmscripten : public Pass {
         // The first operand is the function pointer index, which must be
         // constant if we are to optimize it statically.
         if (auto* index = curr->operands[0]->dynCast<Const>()) {
-          size_t indexValue = index->value.getInteger();
```
Comment (sbc100) [inline src/passes/PostEmscripten.cpp:320]: This is just a drive by fix, that I noticed while fixing the line above.

Function pointer are also unsigned values.

**Comment by sbc100** [comment]
I added a test!

File: test/lit/passes/post-emscripten.wast [hunk 2]
```diff
@@ -40,4 +47,11 @@
  (export "__start_foo" (global $foo_start))
  ;; CHECK:      (export "__stop_foo" (global $foo_stop))
  (export "__stop_foo" (global $foo_stop))
+ (func $meminit
+  (memory.init $mem $data2
+   (i32.const 3000000000)
+   (i32.const 0)
+   (i32.const 21)
+  )
+ )
```
Comment (kripken) [inline test/lit/passes/post-emscripten.wast:56]: Why does this function not appear in the CHECKs?

**Review by kripken — COMMENTED** [review]
I see now, thanks. lgtm % question about the test.

File: test/lit/passes/post-emscripten.wast (hunk 2 above)
Comment (sbc100) [reply test/lit/passes/post-emscripten.wast:56]: Oh.. i need to re-run the auto-generater.

**Pull request merged**